src/sage/algebras/lie_algebras/superliealgebra.py

```diff
@@ -0,0 +1,354 @@
+"""
+Super Lie Algebras
+
+These are Z-graded Lie algebras satisfying graded antisymmetry and the graded Jacobi identity.
```

Comment:
Currently the testing doesn't seem to check that the Z grading is satisfied. Is this intended? If so, this documentation will need to be adjusted.

More generally, this doc will need to specify that it only handles the finite dimensional case given by structure coefficients.

Reply:
For the current implementation, I'm not entirely sure what kind of test I ought to run to check for Z grading. Maybe just a check that every element in the "degree" input tuple is an integer?

Lastly-could you elaborate a bit on what you mean by adding documentation? As in, what kind of things would you expect to be added? At any rate, I went through everything once to improve its conformity with the Python convention, and added the specification you mentioned just above.